legend-shared-utility-interfaces/src/main/java/org/finos/legend/shared/test/junit/RunUsingCICDServer.java

```diff
@@ -0,0 +1,24 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.shared.test.junit;
+
+/*
+    Marker interface to be used with Junit Categories. See https://junit.org/junit4/javadoc/4.12/org/junit/experimental/categories/Categories.html
+
+    Used to indicate a test that can only be run on a CI/CD server because that test requires configuration (e.g. secrets) that are injected by the CI/CD server.
+ */
+public interface RunUsingCICDServer
```

Comment:
Why we need this one?  My worry is that extending this interface could lead executing wrong categories.  Like, do we will ever enable RunUsingCICDServer or we will enable RunUsingGithubActions and/or RunUsingTeamCity?

Reply:
Not everyone uses Github or TeamCity. But we can always extract a base interface .

Reply:
But if this is for the platform development, we should target what we use.  Are aiming for people to use these generically?  Even if that is the case, I will avoid extending this interface on the other two categories.